lib/super_digit.rb

```diff
@@ -1,15 +1,22 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def super_digit(n)
-   
-end
+  return n if n < 10
   
+  sum = 0
+  while n > 0
+    sum += n % 10
+    n /= 10
+  end
+
+  return super_digit(sum)
+end
+
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - Is this O(2^n) (exponential), because the method gets called twice each time?
+# Space Complexity - Is this O(2^n) (exponential), because the method gets called twice each time?
```

Comment:
This isn't exponential since the number of digits goes down by 1 with each iteration.